.github/workflows/post-merge.yml

```diff
@@ -58,11 +79,33 @@ jobs:
   #   secrets:
   #     aws-role-arn: ${{ secrets.GHA_AWS_ROLE_ARN }}
 
+  publish-test-image:
+    uses: ./.github/workflows/publish-test-image.yml
+    needs:
+      # - acceptance-test
+      - preview-main
+      - detect-changes
+    if: needs.detect-changes.outputs.acceptance-tests-changed == 'true'
+    permissions:
+      contents: read
+      id-token: write
+      packages: read
+    with:
+      environment: build
+    secrets:
+      aws-role-arn: ${{ secrets.GHA_AWS_ROLE_ARN }}
+      ecr-repository: ${{ secrets.TEST_IMAGE_ECR_REPOSITORY }}
+
   publish:
     uses: ./.github/workflows/publish.yml
     needs:
+      - publish-test-image
       # - acceptance-test
       - preview-main
+    if: |
+      always() &&
```

Comment:
FYI for reviewers ... adding the `if` statement allows you to override the behaviour of `needs` and allow for continuation if one of the `needs` stepped is skipped 😄